### PR TITLE
NEW : feature to link user to contacts like contacts with third parties

### PR DESCRIPTION
**Pull request by aconits**
This feature allow us to link one user to many contacts.
I have duplicate the link between contacts and third parties.

### CODE REVIEW

File: htdocs/install/mysql/migration/3.6.0-3.7.0.sql [hunk 1]
```diff
@@ -1182,3 +1182,13 @@ insert into llx_c_tva(rowid,fk_pays,taux,recuperableonly,localtax1,localtax1_typ
 ALTER TABLE llx_livraison MODIFY COLUMN date_delivery DATETIME NULL DEFAULT NULL;
 
 INSERT INTO llx_const (name, value, type, note, visible, entity) values ('PRODUCT_USE_OLD_PATH_FOR_PHOTO','1','chaine','Use old path for products images',1,1);
+
+-- Contact sales representatives
```
Comment (aspangaro) [inline htdocs/install/mysql/migration/3.6.0-3.7.0.sql]: Hi,

You are in 3.6.0-3.7.0 migration script. 3.7.0 is frozen for new function. Maybe in 3.8.0 ?

Comment (fappels) [inline htdocs/install/mysql/migration/3.6.0-3.7.0.sql]: Better to use llx_element_contact and create a new element 'user' in llx_c_type_contact.

**Comment by simnandez** [comment]
@atm-ph please move the changements from 3.6.0-3.7.0 to 3.7.0-3.8.0

**Comment by eldy** [comment]
What is common reason for this feature ? Exemple of use case ?

**Comment by aconits** [comment]
This is usefull if you want to know to whom belongs a contact and we will next add a right corresponding to this feature to allow a user to see only his contacts. 

Use case is when a customer (thirdparty) is shared by multiple salesrep but the associated contacts are not.

**Comment by rdoursenaud** [comment]
This sounds cumbersome and AFAIK there's already something answering this use case: private contacts.

**Comment by eldy** [comment]
Currently, the author/owner of a contact is already saved into contact record. But this means only one user is linked to contact. If i understood, you would like to have contact to "belong" to several users.

I'am afraid this feature will create a big confusion into how permission works and how current link bewteewn thirdparties and user (link "sale representative") works. The user that "owns" contact may have a setup that is a conflict with what he "owns" as thirdparties of contact.

I understand such feature may be interesting for some use case but i would prefer to see it into an optional external module, due to the way it will break a lot of current mecanism (permissions, what to do if user is linked to contact x belonging to thirdparty y but user is not sale representative of thirdparty y, and more...). A big headach may appears so i would prefer to have such links done and managed by an external module.

**Pull request closed without merging**